homeassistant/components/shelly/entity.py

```diff
@@ -509,13 +513,16 @@ def __init__(
         key: str,
         attribute: str,
         description: RpcEntityDescription,
+        domain: str | None = None,
     ) -> None:
         """Initialize sensor."""
         super().__init__(coordinator, key)
         self.attribute = attribute
         self.entity_description = description
 
-        self._attr_unique_id = f"{super().unique_id}-{attribute}"
+        self._attr_unique_id = f"{super().unique_id}"
+        if domain != Platform.SWITCH:
+            self._attr_unique_id += f"-{attribute}"
```

Comment:
We shouldn't have platform specific logic in the base class. We should instead override the value in the sub class

Reply:
Same as above

Reply:
Maybe my comments on our discord group was lost so I am putting a short summary here:

We have classes which are used by main device platforms (Switch, Cover, Light) - `ShellyBlockEntity` & `ShellyRpcEntity` these classes are used to create main device sensors.

We have classes which are used to create sub device attribute sensors which are used by binary & sensor platforms - `ShellyBlockAttributeEntity` & `ShellyRpcAttributeEntity` these are subclasses of `ShellyBlockEntity` & `ShellyRpcEntity`.

This PR try to take the `ShellyBlockAttributeEntity` & `ShellyRpcAttributeEntity` and revert some of the modifications by the subclass to make it behave similar to `ShellyBlockEntity` & `ShellyRpcEntity`, this approach is against any coding standard and will always fail since we can't align all the entities with this approach without breaking the entity unique ID and name.

Subclass as suggested by @bdraco is basically using the `ShellyBlockEntity` & `ShellyRpcEntity` with some other modifications from `ShellyBlockAttributeEntity` & `ShellyRpcAttributeEntity` which will work but will make 3 classes for every type (or even 6 if we include sleeping platforms) which will make too much redundant code.

I have started to think of a clean approach that will not require to much code modification and duplication (and for sure not using platform specific logic as mentioned above) but it will take time.

Reply:
Was not lost was applied already:


https://github.com/home-assistant/core/blob/09b8e4368f15c1dd8fa890b15ee8e09a28eb324b/homeassistant/components/shelly/switch.py#L197
https://github.com/home-assistant/core/blob/09b8e4368f15c1dd8fa890b15ee8e09a28eb324b/homeassistant/components/shelly/switch.py#L239

I think you are commenting without looking at the code lately.